Source/objects.cpp

```diff
@@ -2266,10 +2266,19 @@ void OperatePedestal(Player &player, Object &pedestal, bool sendmsg)
 		return;
 	}
 
-	if (pedestal._oVar6 == 3 || !RemoveInventoryItemById(player, IDI_BLDSTONE)) {
+	if (pedestal._oVar6 == 3 || (sendmsg && !RemoveInventoryItemById(player, IDI_BLDSTONE))) {

```

Comment:
`OperatePedestal` is called from two call-sites (`OperateObject` and `SyncOpObject`).
`OperateObject` is also called for remote clients in `CheckNewPath`.
In this case `sendmsg` is false but we still need to check if the item is in the inventory, cause it's not validated (or alternative do nothing in `OperateObject` for remote players).
So we must distinguish if `OperatePedestal` got called from `SyncOpObject` or `OperateObject`. That's why other functions got two parameters (`sendmsg` and `sendLootMsg`).
I think we need to do the same here.

This is another issue where multiplayer logic is complex. 😕 
In the long run we should avoid running `OperateObject` for remote players (through `CheckNewPath`) and should always rely on `SyncOpObject`. But this is far outside of the scope of this pr.

Reply:
This change prevents `OperateObject()` from calling `OperatePedestal()` when `sendmsg = false`, and also prevents `SyncOpObject()` from calling `OperatePedestal()` if `sendmsg = true`. Only the local player will invoke it from `CheckNewPath() > OperateObject() > OperatePedestal()` and only remote players will invoke it via `OnOperateObject() > SyncOpObject() > OperatePedestal()`. If you try to handle it in `OperatePedestal()` with two boolean values, you'll end up with something that doesn't make much sense logically, like...

```c++
if (sendmsg != sendLootMsg) {
    return;
}
```

Reply:
Sorry I missed the `if (sendmsg)` check in `OperateObject`. 😢 
This goes in the way that `OperateObject` is not used for remote players that's a nice direction imho.

Reply:
> (or alternative do nothing in `OperateObject` for remote players)

To be clear, if the local player processes their own `CMD_OPERATEOBJ` commands, the whole quest would be finished immediately because he wouldn't be checking for Blood Stone. If remote players process the local player's `CheckNewPath()`, then the game could end up advancing the quest twice depending on the race condition I mentioned in the issue. IMO, your proposed alternative is the only reasonable approach here, and that's what this PR does.